projects/sandbox/export/export.py

```diff
@@ -121,32 +130,36 @@ def export(
         # a concurrent instance count, scale the existing
         # instance group to this value
         if instances is not None:
-            # TODO: should quiver handle this under the hood?
-            try:
-                bbhnet.config.scale_instance_group(instances)
-            except ValueError:
-                bbhnet.config.add_instance_group(count=instances)
+            scale_model(bbhnet, instances)
     except KeyError:
         # otherwise create the model using the indicated
         # platform and set up an instance group with the
         # indicated scale if one was provided
         bbhnet = repo.add("bbhnet", platform=platform)
         if instances is not None:
-            bbhnet.config.add_instance_group(count=instances)
-
-    # turn off graph optimization because of this error
-    # https://github.com/triton-inference-server/server/issues/3418
-    bbhnet.config.optimization.graph.level = -1
+            scale_model(bbhnet, instances)
 
     # export this version of the model (with its current
     # weights), to this entry in the model repository
-    input_shape = (1, num_ifos, int(kernel_length * sample_rate))
+    input_shape = (batch_size, num_ifos, int(kernel_length * sample_rate))
+
+    # TODO: hardcoding these kwargs for now, but worth
+    # thinking about a more robust way to handle this
+    kwargs = {}
+    if platform == qv.Platform.ONNX:
+        kwargs["opset_version"] = 13
+
+        # turn off graph optimization because of this error
+        # https://github.com/triton-inference-server/server/issues/3418
+        bbhnet.config.optimization.graph.level = -1
+    elif platform == qv.Platform.TENSORRT:
+        kwargs["use_fp16"] = True
```

Comment:
Unfortunately it doesn't right now, since it relies on annotations to infer the type the parser needs to map to. A few markup parsers (yaml, toml) have functionality for applying heuristics that we could maybe lift, but no idea how complicated that might turn out to be. Off the top of my head, the real headaches would be
- reimplementing argparse's logic in a less structured format for iterating through the unused arguments and grouping them into `(arg_name, value)` pairs, accounting for booleans which would just be formatted as `--flag-name` with no associated values.
- handling arguments which are misspelled, silently giving you their default behavior (if they don't have a default, this will luckily raise a missing argument error)